### PR TITLE
Create new base image without geosupport for Pluto build

### PR DESCRIPTION
**Pull request by fvankrieken**
Couple things here
- I added new lighter-weight base image that this build is using now. See `docker` folder
- reworked shp_export a bit. It both supports an optional third arg for filename but it's needed to be able to pass additional arguments to ogr2ogr (like for reprojection, see devdb calls). Previously I had done something somewhat ugly and just made it so that any argument after 3rd was fed to ogr2ogr, while if you didn't want to specify a separate filename (from the table name), the third optional arg could just be "-" (or the table name repeated). I tweaked it so that the filename optional arg must be flagged, and still 3rd/4th. So `shp_export my_table POINT -f myfile -t_srs "EPSG:2263"` or `shp_export my_table POINT -t_srs "EPSG:2263"` both work. Still a tad inflexible but I think a little less clumsy

[Latest pluto build](https://github.com/NYCPlanning/data-engineering/actions/runs/5487188300)

[Latest devdb build](https://github.com/NYCPlanning/data-engineering/actions/runs/5481639633) (to make sure shapefile logic all good)

### CODE REVIEW

**Comment by fvankrieken** [comment]
Added a new "actual" base image, without geosupport, as well as the geosupport image

New base image has size of 650 MB, going to see if I can get it down

File: bash/utils.sh [hunk 1]
```diff
@@ -252,10 +252,24 @@ function compress {
...
+
+
 function upload {
     local dataset_name=${1}
     local subfolder=${2} # typically version
-    local branchname=$(git rev-parse --symbolic-full-name --abbrev-ref HEAD)
+    local root_folder=$(find-up .git)
+    local wd=$(pwd)
+    cd root_folder
+    local branchname=$(cat .git/HEAD | cut -d '/' -f3)
```
Comment (fvankrieken) [inline bash/utils.sh:271]: Removed dependency on git for checking branch! This would have solved the devdb issue as well.

**Comment by fvankrieken** [comment]
Building with more lightweight image! Though now having the error seen in devdb so this PR is blocked by #32

**Comment by fvankrieken** [comment]
Running one last build, sorry for early review tag, y'all can wait till this afternoon

https://github.com/NYCPlanning/data-engineering/actions/runs/5477153075

File: .github/workflows/developments_build.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ jobs:
   build:
     runs-on: ubuntu-22.04
     container: 
-      image: nycplanning/base:latest
+      image: nycplanning/build-geosupport:latest
```
Comment (fvankrieken) [inline .github/workflows/developments_build.yml:21]: This image is identical, just more specific name now

**Comment by fvankrieken** [comment]
Actuallly @damonmcc some food for thought - this uses ghcr.io/osgeo/gdal:ubuntu-small-3.6.4 as base image for the new container. The main point of this is to have gdal setup... but really I don't think we need anything for what we're doing other than `gdal-bin`. 

The bigger build image uses our latest geosupport image, which uses python:3.9-slim as it's base image. For the sake of consistency, I should probably start with that and add what we need from there, right? Also at some point worth upgrading that.

Really actually, maybe the docker-geosupport repo should be moved into this docker folder, so all of these can be managed together (and bound to the same base image, versioned in similar ways, etc)

**Comment by fvankrieken** [comment]
@alexrichey note that docker images are changing (for FacDB assuming you're using in build), though I haven't deleted the original "base" one yet

**Review by damonmcc — CHANGES_REQUESTED** [review]

File: db-pluto/pluto_build/bash/config.sh [hunk 3]
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 source ../../bash/utils.sh # assumes being run from pluto_build folder
 
+set_env ./version.env
+
```
[the remark this answers is not in the document]
Comment (fvankrieken) [reply db-pluto/pluto_build/bash/config.sh:5]: You are correct

[the remark this answers is not in the document]
Comment (fvankrieken) [reply db-pluto/pluto_build/bash/config.sh:6]: Fixed

**Review by damonmcc — APPROVED** [review]

**Pull request merged**